### PR TITLE
Converted probitRegDemo to python

### LINKED ISSUE #208: Convert probitRegDemo to python
This is figure number 12.2 in the book. its matlab code is [here](https://github.com/probml/pmtk3/blob/master/demos/probitRegDemo.m)

### PR DESCRIPTION
**Pull request by always-newbie161**
@murphyk @mjsML 

Closes #208

I tried to implement Probit Regression using EM and Minfunc(uses bfgs(or quasi-newton))  for log-likelihood estimation. I used the book and the matlab code as a guide. 

But I think LL is not converging properly😅(probably in minfunc). I request you to review this code and give suggestions 

Screenshot 2021-04-02 at 11 38 05 AM

Screenshot 2021-04-02 at 11 37 50 AM

### CODE REVIEW

**Review by murphyk — CHANGES_REQUESTED** [review]
Line 1. I would add references to the matlab source files (which are a bit hard to find!)
https://github.com/probml/pmtk3/blob/master/toolbox/SupervisedModels/probitReg/sub/probitRegFitEm.m
https://github.com/probml/pmtk3/blob/master/toolbox/SupervisedModels/probitReg/sub/probitRegFitMinfunc.m

Line 27. `value` seems ununsed.

Line 28. This is the cross entropy y*log(q) + (1-y)*log(1-q)?
Your math for the second term looks wrong. Also, what is np.ma.log?
Also, be careful, my matlab code assumed labels were -1,+1 ("plus-minus1"), instead of 0,1. You don't need to slavishly copy me, you should use 0,1.

Line 64. Don't cal it minfunc - that's the name of the particular matlab optimization package. The method should be called fit_gradient or something.

Line 77. minimize is a gradient free optimizer (zero'th order) - you should use BFGS, and provide the gradient and Hessian as args. I give explicit expressions in the book, but you can also use jax's autodiff to derive them (look at [this colab](https://colab.research.google.com/github/probml/pyprobml/blob/master/notebooks/logreg_jax.ipynb) for inspiration - it fits logistic regression using BFGS, where the gradient/Hessian is computed both by hand and using JAX.)

Line 117. Please compare your answers to https://www.statsmodels.org/stable/generated/statsmodels.discrete.discrete_model.Probit.html. The objective is convex, so all 3 methods should give the same results.

**Comment by always-newbie161** [comment]
@murphyk 

Found my mistakes 

these are the figures
Screenshot 2021-04-04 at 4 33 57 PM

**Review by murphyk — CHANGES_REQUESTED** [review]
This looks good. However, it seems [Probit.fit_regularized](https://www.statsmodels.org/stable/generated/statsmodels.discrete.discrete_model.Probit.fit_regularized.html#statsmodels.discrete.discrete_model.Probit.fit_regularized) uses L1 regularization, which cannot be directly compared to the L2 regularizer used by BFGS. Can you try [Probit.fit](https://www.statsmodels.org/stable/generated/statsmodels.discrete.discrete_model.Probit.fit.html#statsmodels.discrete.discrete_model.Probit.fit) instead, which comptues MLE? (For BFGS, set lambda to be small to approximate MLE)

Also, you define an estep function, but there is no mstep function. I realize the M step is just ridgeReg, but it might help clarity to define mstep() anyway.

Finally, please use [PEP 8 naming conventions](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)

**Comment by always-newbie161** [comment]
@murphyk 

- Used Probit.fit() with bfgs
- Added separate M step function and decreases the lam to 1e-2
- Used PEP 8 naming
